src/optimize.ml

```diff
@@ -0,0 +1,89 @@
+open Types
+open Types.Simplified
+
+let rec optimize_expr (exp : expr) : expr =
+  match exp with
+  | I32_const x :: I32_const y :: I_binop (S32, Add) :: tl ->
+      optimize_expr (I32_const (Int32.add x y) :: tl)
+  | I32_const x :: I32_const y :: I_binop (S32, Sub) :: tl ->
+      optimize_expr (I32_const (Int32.sub x y) :: tl)
+  | I32_const x :: I32_const y :: I_binop (S32, Mul) :: tl ->
+      optimize_expr (I32_const (Int32.mul x y) :: tl)
+  | I32_const x :: I32_const y :: I_binop (S32, Div U) :: tl ->
+      optimize_expr (I32_const (Int32.div x y) :: tl)
+  | I64_const x :: I64_const y :: I_binop (S64, Add) :: tl ->
+      optimize_expr (I64_const (Int64.add x y) :: tl)
+  | I64_const x :: I64_const y :: I_binop (S64, Sub) :: tl ->
+      optimize_expr (I64_const (Int64.sub x y) :: tl)
+  | I64_const x :: I64_const y :: I_binop (S64, Mul) :: tl ->
+      optimize_expr (I64_const (Int64.mul x y) :: tl)
+  | I64_const x :: I64_const y :: I_binop (S64, Div U) :: tl ->
+      optimize_expr (I64_const (Int64.div x y) :: tl)
+  (* | I32_const (Int32.zero) ::  :: tl -> EQZ
```

Comment:
What is this ?

Reply:
I tried to build the following optimization :
.eqz instruction after const 0 is always true